opm/parser/eclipse/Parser/ErrorGuard.hpp

```diff
@@ -0,0 +1,51 @@
+/*
+  Copyright 2019 Joakim Hove/datagr
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef ERROR_GUARD_HPP
+#define ERROR_GUARD_HPP
+
+#include <string>
+#include <vector>
+
+namespace Opm {
+
+class ErrorGuard {
+public:
+    void add_error(const std::string& errorKey, const std::string& msg);
```

Comment:
I think we should stay with the addError() style used almost everywhere in OPM, including opm-common.

Reply:
Hehe - I am fighting to change that - don't you see :-)

Reply:
Yes, and my beef with it is that the camelCase() convention for functions is about the only thing we have managed to get all developers to use consistently (99%).